### PR TITLE
[7.x] Add AuthenticationException on Gate

### PR DESCRIPTION
**Pull request by Oipnet**
**Curently :**
When a user is not authenticated and is required in the policy an http response with 403 is return

**Now :**
When a user is not authenticated and is required in the policy an http response with 401 is return

### CODE REVIEW

**Comment by kadevland** [comment]
it's good feature and more logical

File: src/Illuminate/Auth/Access/Gate.php [hunk 1]
```diff
@@ -705,6 +708,10 @@ protected function callPolicyMethod($policy, $method, $user, array $arguments)
         if ($this->canBeCalledWithUser($user, $policy, $method)) {
             return $policy->{$method}($user, ...$arguments);
         }
+
+        if (! $user) {
+            throw new \Illuminate\Auth\AuthenticationException();
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Access/Gate.php:713]: Please import this (but don't update the phpdoc).

File: tests/Auth/AuthAccessGateTest.php [hunk 2]
```diff
@@ -116,7 +116,12 @@ public function testPoliciesCanAllowGuests()
         $gate->policy(AccessGateTestDummy::class, AccessGateTestPolicyThatAllowsGuests::class);
 
         $this->assertTrue($gate->check('edit', new AccessGateTestDummy));
-        $this->assertFalse($gate->check('update', new AccessGateTestDummy));
+        try {
+            $gate->check('update', new AccessGateTestDummy);
+        } catch (\Exception $e) {
+            $this->assertInstanceOf(\Illuminate\Auth\AuthenticationException::class, $e);
```
Comment (GrahamCampbell) [inline tests/Auth/AuthAccessGateTest.php:122]: Same.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Auth/Access/Gate.php (hunk 1 above)
Comment (Oipnet) [reply src/Illuminate/Auth/Access/Gate.php]: done

File: tests/Auth/AuthAccessGateTest.php (hunk 2 above)
Comment (Oipnet) [reply tests/Auth/AuthAccessGateTest.php]: done

**Comment by taylorotwell** [comment]
Breaking change. May break people's tests, etc.

**Comment by kadevland** [comment]
I understand the possibilite of Breaking change.
May be if policy have config file and in config file set active 
"this authenticated and is required and null  response with 401 "

**Comment by Oipnet** [comment]
@taylorotwell do you think this breaking change can be include in future version like Laravel 8.

**Pull request closed without merging**